src/Controls/src/Core/Platform/AlertManager/AlertManager.iOS.cs

```diff
@@ -206,6 +215,20 @@ static void PresentPopUp(Window virtualView, UIWindow platformView, UIAlertContr
 					_ = platformView.RootViewController.PresentViewControllerAsync(alert, true);
 				});
 			}
+
+			bool PageIsInThisWindow(IView view)
+			{
+				if (view is not Page page)
+					return false;
+
+				var window = page.Window;
+				var platformWindow = window?.MauiContext.GetPlatformWindow();
+
+				if (window is null || platformWindow is null)
+					return false;
```

Comment:
AFAIK we will never be in a situation in which platformWindows is not null and window is null.

Reply:
Replaced all the logic.